### ISSUE TITLE
Deprecate WarpImageFilter and WarpVectorImageFilter

### ISSUE THREAD

**Issue by blowekamp** [InsightSoftwareConsortium/ITK#809]
These two filers apply deformation fields to images ( and vector images ). The same functionality now exist in the `ResampleImageFilter`. 

Should these filters be deprecated to avoid confusion?

**Comment by blowekamp** [comment]
@thewtex @ntustison @hjmjohnson 
Any thought on these classes?

**Comment by hjmjohnson** [comment]
@blowekamp I am in favor of deprecating as long a documentation clearly expresses how to get the same results with ResampleImageFilter (Including an example refactoring).

Thanks Brad.

**Comment by blowekamp** [comment]
Here is list of examples which use the `WarpImageFilter`:
```
Filtering/WarpImageFilter1.cxx
RegistrationITKv3/DeformableRegistration10.cxx
RegistrationITKv3/DeformableRegistration16.cxx
RegistrationITKv3/DeformableRegistration17.cxx
RegistrationITKv3/DeformableRegistration2.cxx
RegistrationITKv3/DeformableRegistration3.cxx
RegistrationITKv3/DeformableRegistration5.cxx
RegistrationITKv3/DeformableRegistration9.cxx
RegistrationITKv3/LandmarkWarping2.cxx
RegistrationITKv4/DeformableRegistration10.cxx
RegistrationITKv4/DeformableRegistration16.cxx
RegistrationITKv4/DeformableRegistration17.cxx
RegistrationITKv4/DeformableRegistration2.cxx
RegistrationITKv4/DeformableRegistration3.cxx
RegistrationITKv4/DeformableRegistration5.cxx
RegistrationITKv4/DeformableRegistration9.cxx
RegistrationITKv4/LandmarkWarping2.cxx
```

Looks like step one is to update those examples to use the `ResampleImageFilter`.

**Comment by blowekamp** [comment]
We need to figure out what to do with the ITKv3 Registration examples...

**Comment by hjmjohnson** [comment]
The ITKv3 examples should definitely be deprecated, and I would recommend removing them (as they are now confusing to users who accidentally end up in that directory.

**Comment by hjmjohnson** [comment]
On second thought, perhaps we should just move them out of the "examples" directory and into a "testing" subdirectory as they may still have code coverage utility.

ITKv6 should evaluate if the ITKv3 framework is completely removed.

**Comment by blowekamp** [comment]
I agree that it is important to keep them as testing while we have the ITKv3 registration framework still there.

Tangentially related, ITKv4 registration framework not sounds old. Perhaps is would have been better to call the old one "legacy" registration framework, and just call the current one "the registration framework". I'm not sure how often it's called ITKv4 registration framework.

**Comment by dzenanz** [comment]
Can this now be closed? There was some work on this topic over the past month.

**Comment by blowekamp** [comment]
There is still quite of number of examples which should be updated.

**Comment by dzenanz** [comment]
I just noticed [compile errors](https://open.cdash.org/viewBuildError.php?buildid=6018132) in non-default module `ITKFEMRegistration`:
`C:\Dashboard\ITK\Modules\Registration\FEM\include\itkFEMRegistrationFilter.h(166): error C2061: syntax error: identifier 'VectorInterpolateImageFunction' [C:\Dashboard\ITK-build\Modules\Registration\FEM\test\ITKFEMRegistrationTestDriver.vcxproj]`. Is this on your radar @blowekamp?

**Comment by blowekamp** [comment]
Nope, I hadn't seen this before. Is it more related to one of my changes or @maekclena changes?

**Comment by dzenanz** [comment]
Failures started on 2019-07-08. The probable culprit is 9ef036a9cbc2a9f42059f4574fdcf4f0fc83bf88, by @maekclena.

**Comment by blowekamp** [comment]
Changing the Examples to use the ResampleImageFilter over WarpImageFilter should be a good first issue.

**Comment by krupalbhat** [comment]
I found only one example using WarpImageFilter, which is WarpImageFilter1.cxx. Are there any other examples that need to be converted?

**Comment by dzenanz** [comment]
There has been work on this over the years. It is possible there is only one left. @blowekamp Maybe that one was left intentionally? If so, we could close this issue.

**Comment by blowekamp** [comment]
Awesome! 🥳  Looks like this is the only one left.

The documentation of this example says: "This example illustrates how to use the WarpImageFilter and a  deformation field for resampling an image."

So is this example needed? Is there an example that uses the ResampleImageFilter to apply a deformation field? 

If there is no existing example than which uses the ResampleImageFilter, then this example will need to be renamed and the purpose of it changed.
